zerver/migrations/0173_user_message_add_is_private_flag.py

```diff
@@ -0,0 +1,45 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.13 on 2018-06-14 13:39
+from __future__ import unicode_literals
+
+import bitfield.models
+from django.db import migrations, transaction
+from django.db.backends.postgresql_psycopg2.schema import DatabaseSchemaEditor
+from django.db.migrations.state import StateApps
+from django.db.models import Q, F
+
+def set_initial_value_of_is_private_flag(
+        apps: StateApps, schema_editor: DatabaseSchemaEditor) -> None:
+    UserMessage = apps.get_model("zerver", "UserMessage")
+    UserProfile = apps.get_model("zerver", "UserProfile")
+    user_profiles = UserProfile.objects.all()
+    for user_profile in user_profiles:
+        with transaction.atomic():
+            private_objects = UserMessage.objects.filter(Q(message__recipient__type = 1) | Q(message__recipient__type = 3),
+                                                         user_profile = user_profile)
+            stream_objects = UserMessage.objects.filter(message__recipient__type = 2,
+                                                        user_profile = user_profile)
+            private_objects.update(flags=F('flags').bitor(UserMessage.flags.is_private))
+            stream_objects.update(flags=F('flags').bitand(~UserMessage.flags.is_private))
```

Comment:
I'm wondering if it would be cheaper to do it like this:

* clear all flags where is_private = True (probably very few, since the old flag was rare)
* set all flags for private_objects

This will result in fewer updates and avoid the second query that has to join to Recipient via Message.